csrc/python_frontend/fusion_record.h

```diff
@@ -2645,107 +2645,80 @@ struct IotaOpRecord : RecordFunctor {
 };
 
 //! Specialized Record Functors for random ops.
-struct RandomOpRecord : RecordFunctor {
-  RandomOpRecord(
+template <serde::RecordType RType>
+struct RandomDistOpRecord : RecordFunctor {
+  RandomDistOpRecord(
       std::vector<State> _args,
       std::vector<State> _outputs,
-      std::vector<State> output_shape,
-      std::string _name,
       PrimDataType dtype)
-      : RecordFunctor(
-            std::move(_args),
-            std::move(_outputs),
-            _name,
-            serde::RecordType_RandomOp),
-        output_shape_(std::move(output_shape)),
+      : RecordFunctor(std::move(_args), std::move(_outputs), "", RType),
         dtype_(dtype) {
-    if (args_.size() == 4) {
-      // seed and offset were provided in addition to the usual 2 arguments
-      setArgName(2, "rng_seed");
-      setArgName(3, "rng_offset");
+    static_assert(
+        (RType == serde::RecordType_NormalDistOp) ||
+        (RType == serde::RecordType_UniformDistOp));
+    if constexpr (RType == serde::RecordType_UniformDistOp) {
+      name_ = "ops.uniform";
+    } else if constexpr (RType == serde::RecordType_NormalDistOp) {
+      name_ = "ops.normal";
+    }
```

Comment:
maybe assert in the else block for an unrecognized template parameter, instead of the static_assert above.

Reply:
This should be covered by the compile time `static_assert` at the top of the function.

Reply:
I also put `static_assert` in `else` statement.